ntp-daemon/src/peer_manager.rs

```diff
@@ -78,41 +153,69 @@ impl<C: NtpClock> Peers<C> {
         }
     }
 
-    async fn add_peer_internal(&mut self, config: Arc<PeerConfig>) -> JoinHandle<()> {
+    /// Add a single peer to an existing pool
+    async fn add_pool_peer_internal(
+        &mut self,
+        address: NormalizedAddress,
+        mut cached: CachedPoolAddresses,
+    ) -> JoinHandle<()> {
         let index = self.indexer.get();
-        let addr = loop {
-            let host = match &*config {
-                PeerConfig::Standard(StandardPeerConfig { addr, .. }) => {
-                    debug!(unresolved = ?&addr, "lookup host");
-                    addr.lookup_host().await.map(|mut i| i.next())
-                }
-
-                PeerConfig::Pool(PoolPeerConfig { addr, .. }) => {
-                    debug!(unresolved = ?&addr, "lookup host");
-                    addr.lookup_host().await.map(|mut i| i.next())
-                }
-            };
 
-            match host {
-                Ok(Some(addr)) => {
-                    debug!(resolved=?addr, "resolved peer");
-                    break addr;
-                }
-                Ok(None) => {
-                    warn!("Could not resolve peer address, retrying");
-                    tokio::time::sleep(NETWORK_WAIT_PERIOD).await
-                }
-                Err(e) => {
-                    warn!(error = ?e, "error while resolving peer address, retrying");
-                    tokio::time::sleep(NETWORK_WAIT_PERIOD).await
+        // socket addresses of the peers of this pool that are currently active
+        let active_pool_peers = self.peers.values().filter_map(|p| match &p.peer_address {
+            PeerAddress::Peer { .. } => None,
+            PeerAddress::Pool {
+                address: peer_address,
+                socket_address,
+                ..
+            } => {
+                if &address == peer_address {
+                    Some(*socket_address)
+                } else {
+                    None
```

Comment:
Why are we matching on peer address here, instead of just checking whether they use the same peerconfig?

Reply:
this suggests we really want a pool index, similar to how we already have a peer index: some unique identifier for a pool. As you say, the normalized address is not guaranteed to be unique (though it's weird if it isn't)

Reply:
I dont think this needs to be an index per see, if all peers have an arc to the same peerconfig then simple pointer equality on that would suffice. Or perhaps the peerconfig could contain a list of the peerindices associated with it?

Reply:
idk, I implemented the index and I think that works very nicely. Means we don't need additional `Arc` stuff